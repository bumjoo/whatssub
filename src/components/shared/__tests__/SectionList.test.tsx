import * as React from 'react';

// Note: test renderer must be required after react-native.
import styled, { ThemeProvider } from 'styled-components/native';
import SectionList from '../SectionList';
import renderer from 'react-test-renderer';

const SampleItemText = styled.Text``;
const component = (props?: any, sampeSectionTitle?: string) => {
  return (
    <SectionList
      renderItem={() => (
        <SampleItemText>Sample Item</SampleItemText>
      )}
      sections={[{
        title: sampeSectionTitle || undefined,
        data: [],
      }]}
      {...props}
    />
  );
};

describe('[SectionList]', () => {
  const sampeSectionTitle = 'Sample section title';

  it('should render without crashing', () => {
    const rendered = renderer.create(component());
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  it(`should render [SectionEmptyHeader]"`, () => {
    const rendered = renderer.create(component());
    const sectionHeader = rendered.root.findByProps({
      testID: 'sectionEmptyHeader',
    });

    expect(sectionHeader).toBeTruthy();
  });

  it(`should render [SectionHeader] with title "${sampeSectionTitle}"`, () => {
    const rendered = renderer.create(component({}, sampeSectionTitle));
    const sectionHeader = rendered.root.findByProps({
      testID: 'sectionHeader',
    });
    const sectionHeaderText = sectionHeader.findByProps({
      children: sampeSectionTitle,
    });

    expect(sectionHeaderText).toBeTruthy();
  });
});
