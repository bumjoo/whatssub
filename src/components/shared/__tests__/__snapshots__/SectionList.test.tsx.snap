// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SectionList] should render without crashing 1`] = `
<RCTScrollView
  ListFooterComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "displayName": "Styled(View)",
      "inlineStyle": InlineStyle {
        "rules": Array [
          "
  display: flex;
  width: 100%;
  height: 1px;
  background-color: rgb(230, 234, 238);
",
        ],
      },
      "render": [Function],
      "styledComponentId": "StyledNativeComponent",
      "target": [Function],
      "withComponent": [Function],
    }
  }
  data={
    Array [
      Object {
        "data": Array [],
        "title": undefined,
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  renderSectionHeader={[Function]}
  scrollEventThrottle={50}
  sections={
    Array [
      Object {
        "data": Array [],
        "title": undefined,
      },
    ]
  }
  stickyHeaderIndices={
    Array [
      0,
    ]
  }
  stickySectionHeadersEnabled={true}
  style={
    Array [
      Object {
        "backgroundColor": "rgb(249, 250, 251)",
      },
    ]
  }
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(230, 234, 238)",
              "display": "flex",
              "height": 1,
              "width": "100%",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(249, 250, 251)",
              "display": "flex",
              "height": 12,
              "justifyContent": "center",
              "paddingBottom": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 0,
            },
          ]
        }
        testID="sectionEmptyHeader"
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(230, 234, 238)",
              "display": "flex",
              "height": 1,
              "width": "100%",
            },
          ]
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
